src/spyglass/lfp/lfp_electrode.py

```diff
@@ -33,30 +38,85 @@ def create_lfp_electrode_group(
+
+        Raises
+        ------
+        ValueError
+            If the session is not found in the Session table or
+            if the electrode list is empty or
+            if the electrodes are not valid for this session.
         """
-        # remove the session and then recreate the session and Electrode list
-        # check to see if the user allowed the deletion
-        key = {
+        # Validate inputs
+        session_key = {"nwb_file_name": nwb_file_name}
+        if not (Session() & session_key):
+            raise ValueError(
+                f"Session '{nwb_file_name}' not found in Session table."
+            )
+
+        if not isinstance(electrode_list, (list, np.ndarray)):
+            raise ValueError(
+                "electrode_list must be a list or numpy array of integers."
+            )
+
+        if len(electrode_list) == 0:
+            raise ValueError("electrode_list cannot be empty.")
+
+        if isinstance(electrode_list, np.ndarray):
+            # convert to list[int] if numpy array
+            electrode_list = electrode_list.astype(int).ravel().tolist()
+
+        # Sort and remove duplicates
+        electrode_list = sorted(list(set(electrode_list)))
+
+        # Check against valid electrodes for this session in the database
+        electrode_table = Electrode() & session_key
+        if not electrode_table:
+            raise ValueError(
+                f"No electrodes found for session '{nwb_file_name}'."
+            )
+        if np.any(
+            np.isin(
+                electrode_list,
+                electrode_table.fetch("electrode_id"),
+                invert=True,
+            )
+        ):
+            raise ValueError(
+                f"Invalid electrode_id(s) provided for "
+                f"nwb_file_name '{nwb_file_name}'. They do not exist in the "
+                f"Electrode table for this session."
+            )
+
+        master_key = {
             "nwb_file_name": nwb_file_name,
             "lfp_electrode_group_name": group_name,
         }
-        LFPElectrodeGroup().insert1(key, skip_duplicates=True)
+        electrode_restrictions = [
+            {"electrode_id": eid} for eid in electrode_list
+        ]
+        electrode_keys_to_insert = (
+            electrode_table & electrode_restrictions
+        ).fetch("KEY")
+        part_keys = [
+            {**master_key, **electrode_key}
+            for electrode_key in electrode_keys_to_insert
+        ]
+
+        # Insert within a transaction for atomicity
+        # (Ensures master and parts are inserted together or not at all)
+        connection = LFPElectrodeGroup.connection
+        with connection.transaction:
+            # Insert master table entry (skips if already exists)
+            LFPElectrodeGroup().insert1(master_key, skip_duplicates=True)
+            # Insert part table entries (skips duplicates)
+            LFPElectrodeGroup.LFPElectrode.insert(
+                part_keys, skip_duplicates=True
```

Comment:
If you don't expect duplicates, I would remove - it seems like they would be avoided by the input validation above - sorted set line